### PR TITLE
Fix disclaiming logic to include code cache segments

### PR DESCRIPTION
**Pull request by dsouzai**
This PR fixes the logic that ensures file backed memory to include code cache segments. `allocateFixedMemorySegmentInList` is not invoked when allocating the code cache repository; `allocateMemoryForSegment` is. However, `allocateMemoryForSegment` is also invoked by `allocateFixedMemorySegmentInList`. Therefore, moving the logic to allocate file backed memory to `allocateMemoryForSegment` covers all memory types that need it.

### CODE REVIEW

**Comment by dsouzai** [comment]
@mpirvu could you please review?

File: runtime/vm/segment.c [hunk 1]
```diff
@@ -240,6 +240,17 @@ allocateMemoryForSegment(J9JavaVM *javaVM,J9MemorySegment *segment, J9PortVmemPa
 		 */
 		Assert_VM_true(J9_ARE_NO_BITS_SET(segment->type, MEMORY_TYPE_VIRTUAL) || J9_ARE_ANY_BITS_SET(segment->type, ~MEMORY_TYPE_VIRTUAL));
 
+		/* Memory segments can be marked as MEMORY_TYPE_DISCLAIMABLE_TO_FILE
+		 * to indicate the intent of allocating memory backed-up by a file.
+		 * This information is passed to the omr port library by setting the
+		 * flag OMRPORT_VMEM_MEMORY_MODE_SHARE_TMP_FILE_OPEN.
+		 * Note that when setting when the OMRPORT_VMEM_MEMORY_MODE_SHARE_TMP_FILE_OPEN bit,
+		 * we also have to set the OMRPORT_VMEM_MEMORY_MODE_SHARE_FILE_OPEN bit.
```
Comment (mpirvu) [inline runtime/vm/segment.c:248]: Small typo coming from the original code: "when" is repeated.

**Review by mpirvu — APPROVED** [review]
LGTM.

File: runtime/vm/segment.c (hunk 1 above)
Comment (dsouzai) [reply runtime/vm/segment.c:248]: Fixed in force push.

**Comment by mpirvu** [comment]
jenkins test sanity all jdk21

**Comment by dsouzai** [comment]
zlinux failure due to https://github.com/eclipse-openj9/openj9/issues/19027
```
[2024-06-18T18:09:42.025Z] Testing: Create and Restore Criu Checkpoint Image once - MethodTypeDeadlockTest
[2024-06-18T18:09:42.025Z] Test start time: 2024/06/18 14:09:41 Eastern Standard Time
[2024-06-18T18:09:42.025Z] Running command: bash /home/jenkins/workspace/Test_openjdk21_j9_sanity.functional_s390x_linux_Personal_testList_1/aqa-tests/TKG/../../jvmtest/functional/cmdLineTests/criu/criuScript.sh /home/jenkins/workspace/Test_openjdk21_j9_sanity.functional_s390x_linux_Personal_testList_1/aqa-tests/TKG/../../jvmtest/functional/cmdLineTests/criu /home/jenkins/workspace/Test_openjdk21_j9_sanity.functional_s390x_linux_Personal_testList_1/jdkbinary/j2sdk-image/bin/java " -XX:+JVMPortableRestoreMode  -XX:+ThrowOnDelayedCheckpointOperation -XX:sleepMillisecondsForNotCheckpointSafe=20 -Xtrace:print=j9vm.731 --add-opens java.base/jdk.internal.misc=ALL-UNNAMED  --add-exports java.base/openj9.internal.criu=ALL-UNNAMED" org.openj9.criu.DeadlockTest MethodTypeDeadlockTest 1 false false
[2024-06-18T18:09:42.025Z] Time spent starting: 10 milliseconds
[2024-06-18T18:09:45.341Z] Time spent executing: 4116 milliseconds
[2024-06-18T18:09:45.341Z] Test result: FAILED
[2024-06-18T18:09:45.341Z] Output from test:
[2024-06-18T18:09:45.341Z]  [OUT] start running script
[2024-06-18T18:09:45.342Z]  [OUT] export GLIBC_TUNABLES=glibc.cpu.hwcaps=-XSAVEC,-XSAVE,-AVX2,-ERMS,-AVX,-AVX_Fast_Unaligned_Load
[2024-06-18T18:09:45.342Z]  [OUT] export LD_BIND_NOT=on
[2024-06-18T18:09:45.342Z]  [OUT] /home/jenkins/workspace/Test_openjdk21_j9_sanity.functional_s390x_linux_Personal_testList_1/jdkbinary/j2sdk-image/bin/java -XX:+EnableCRIUSupport  -XX:+JVMPortableRestoreMode  -XX:+ThrowOnDelayedCheckpointOperation -XX:sleepMillisecondsForNotCheckpointSafe=20 -Xtrace:print=j9vm.731 --add-opens java.base/jdk.internal.misc=ALL-UNNAMED  --add-exports java.base/openj9.internal.criu=ALL-UNNAMED -cp /home/jenkins/workspace/Test_openjdk21_j9_sanity.functional_s390x_linux_Personal_testList_1/aqa-tests/TKG/../../jvmtest/functional/cmdLineTests/criu/criu.jar org.openj9.criu.DeadlockTest MethodTypeDeadlockTest 1
[2024-06-18T18:09:45.342Z]  [OUT] Pre-checkpoint
[2024-06-18T18:09:45.342Z]  [OUT] main: Tue Jun 18 14:09:41 EDT 2024, Performing CRIUSupport.checkpointJVM(), System.currentTimeMillis(): 1718734181491, System.nanoTime(): 1718734181486345517
[2024-06-18T18:09:45.342Z]  [OUT] JVMJITM044W Some or all compiled code in the code cache invalidated post restore.
[2024-06-18T18:09:45.342Z]  [OUT] initiate restore
[2024-06-18T18:09:45.342Z]  [OUT] Error (criu/protobuf.c:72): Unexpected EOF on (empty-image)
[2024-06-18T18:09:45.342Z]  [OUT] Error (criu/arch/s390/crtools.c:776): Cannot open core for virt pid 1293490: No such file or directory
[2024-06-18T18:09:45.342Z]  [OUT] Error (criu/cr-restore.c:2142): Restoring regs for 1293490 failed: No such file or directory
[2024-06-18T18:09:45.342Z]  [OUT] Error (criu/cr-restore.c:2520): Killing processes because of failure on restore.
[2024-06-18T18:09:45.342Z]  [OUT] The Network was unlocked so some data or a connection may have been lost.
[2024-06-18T18:09:45.342Z]  [OUT] Error (criu/cr-restore.c:2547): Restoring FAILED.
[2024-06-18T18:09:45.342Z]  [OUT] Removed test output files
[2024-06-18T18:09:45.342Z]  [OUT] finished script
[2024-06-18T18:09:45.342Z]  [ERR] /home/jenkins/workspace/Test_openjdk21_j9_sanity.functional_s390x_linux_Personal_testList_1/aqa-tests/TKG/../../jvmtest/functional/cmdLineTests/criu/criuScript.sh: line 41: 1293462 Killed                  $2 -XX:+EnableCRIUSupport $3 -cp "$1/criu.jar" $4 $5 $6 > testOutput 2>&1
```

**Pull request merged**